packages/react/lib/components.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+/* global __webpack_modules__, __webpack_require__ */
+
+const { createElement, Component } = require('react');
+const { default: withRouter } = require('react-router-dom/es/withRouter');
+
+exports.Miss = withRouter(({ staticContext }) => {
+  if (staticContext) {
+    staticContext.miss = true;
+  }
+  return null;
+});
+
+exports.Status = withRouter(({ staticContext, code }) => {
+  if (staticContext) {
+    staticContext.status = code;
+  }
+  return null;
+});
+
+exports.Header = withRouter(({ staticContext, name, value }) => {
+  if (staticContext) {
+    staticContext.headers = { ...staticContext.headers, [name]: value };
+  }
+  return null;
+});
+
+exports.Import = ({ module, load, weakId }, name = 'default') => {
+  class ImportPlaceholder extends Component {
+    constructor({ load, name }) {
```

Comment:
Why use the constructor and not `componentWillMount` or `componentDidMount`? If I understand this correctly, the `ImportPlaceholder` will never be used on the server anyways, right?